GSA-TTS/FAC
# Supporting > 10k federal awards

Addresses https://github.com/GSA-TTS/FAC/issues/3529

In this PR:
- Updating `build_templates` to look for `MAX_ROWS` env var, this way we can create workbooks with 20k rows
- We've determined that uploading workbooks > 10k rows only has issues with federal award # entries being greater than 4 digits. Because of this, we're updating regexes/schemas to allow federal award #s to have 4 or 5 digits, with the plan to only allow 5 at some point down the road.
- Adding a bunch of workbooks to `/should_pass` and `/should_fail` to cover these cases
- Increasing upload timeout to 4m, as it took ~3m to upload the 20k line workbook in preview
- It seems `make build_templates` wasn't run after changing an error message for `FederalProgramNameValidation`, so I guess that's being pulled in too

Testing:
- Generating workbooks
  - cd into `/schemas`
  - `make build_templates` should still create a 10k row workbook
  - `MAX_ROWS=100 make build_templates` should create a 100 row workbook
  - `MAX_ROWS=asdf make build_templates` should still create a 10k row workbook
- Upload testing
  - Start a submission
  - Normal federal awards and audit findings uploads should work (the ones in `audit/fixtures/workbooks/should_pass/134732-21/` are sufficient)
  - Uploading a [20k federal awards workbook](https://github.com/GSA-TTS/FAC/files/14723167/Copy.of.federal-awards-workbook-lots-of-rows.xlsx) should work without timing out
  - Modify one of the federal award numbers to have 5 digits (such as `AWARD-0041` to `AWARD-00410`). Both uploads should fail, since all must have the same length.
  - Modify all federal award numbers to have 5 digits. Uploading these should validate.
  - Upload federal award and audit findings workbooks such that one of them uses 4 digit numbers, and the other uses 5 digits. Cross validation should fail here. Uploading ones with both of length 4 or both of length 5 should pass.

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
